### ISSUE TITLE
Unable to download from bbcamerica

### ISSUE THREAD

**Issue by keybounce** [rg3/youtube-dl#10488]
## Please follow the guide below

- You will be asked some questions and requested to provide some information, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *issue* (like that [x])
- Use *Preview* tab to see how your issue will actually look like

---

### Make sure you are using the *latest* version: run `youtube-dl --version` and ensure your version is *2016.08.28*. If it's not read [this FAQ entry](https://github.com/rg3/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl) and update. Issues with outdated version will be rejected.
- [x ] I've **verified** and **I assure** that I'm running youtube-dl **2016.08.28**

### Before submitting an *issue* make sure you have:
- [x ] At least skimmed through [README](https://github.com/rg3/youtube-dl/blob/master/README.md) and **most notably** [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [x ] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones

### What is the purpose of your *issue*?
- [x ] Bug report (encountered problems with youtube-dl)
- [ x] Site support request (request for adding support for a new site)
- [ ] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

---

### The following sections concretize particular purposed issues, you can erase any section (the contents between triple ---) not applicable to your *issue*

---

### If the purpose of this *issue* is a *bug report*, *site support request* or you are not completely sure provide the full verbose output as follows:

Add `-v` flag to **your command line** you run youtube-dl with, copy the **whole** output and insert it here. It should look similar to one below (replace it with **your** log inserted between triple ```):
```
keybounceMBP:Episodes michael$ youtube-dl -v -F http://www.bbcamerica.com/shows/the-hunt/season-1/episode-01-the-hardest-challenge
[debug] System config: []
[debug] User config: [u'-o', u'%(title)s.%(ext)s', u'-f', u'\nbest[ext=mp4][height>431][height<=480]/\nbest[ext=mp4][height>431][height<=576]/\nbestvideo[ext=mp4][height=480]+bestaudio[ext=m4a]/\nbest[ext=mp4][height>340][height<=431]/\nbestvideo[ext=mp4][height>360][height<=576]+bestaudio/\nbest[height>340][height<=576]/\nbestvideo[height>360][height<=576]+bestaudio/\nbestvideo[height=360]+bestaudio/\nbest[ext=mp4][height>=280][height<=360]/\nbest[height<=576]/\nworst', u'--write-sub', u'--write-auto-sub', u'--sub-lang', u'en,enUS', u'--sub-format', u'srt/ass/best', u'--embed-subs', u'--recode-video', u'mp4', u'--mark-watched']
[debug] Command-line args: [u'-v', u'-F', u'http://www.bbcamerica.com/shows/the-hunt/season-1/episode-01-the-hardest-challenge']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2016.08.28
[debug] Python version 2.7.5 - Darwin-13.4.0-x86_64-i386-64bit
[debug] exe versions: ffmpeg 3.1.1, ffprobe 3.1.1, rtmpdump 2.4
[debug] Proxy map: {}
[AMCNetworks] the-hardest-challenge: Downloading webpage
ERROR: Unable to extract media url; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/Users/michael/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 691, in extract_info
    ie_result = ie.extract(url)
  File "/Users/michael/bin/youtube-dl/youtube_dl/extractor/common.py", line 347, in extract
    return self._real_extract(url)
  File "/Users/michael/bin/youtube-dl/youtube_dl/extractor/amcnetworks.py", line 52, in _real_extract
    media_url = self._search_regex(r'window\.platformLinkURL\s*=\s*[\'"]([^\'"]+)', webpage, 'media url')
  File "/Users/michael/bin/youtube-dl/youtube_dl/extractor/common.py", line 650, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
RegexNotFoundError: Unable to extract media url; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.

keybounceMBP:Episodes michael$ youtube-dl -U
youtube-dl is up-to-date (2016.08.28)
keybounceMBP:Episodes michael$ 
```

Attempting to download from BBCAmerica activates the AMCNetwork code, which then fails.

**Comment by remitamine** [comment]
you are trying to download from the wrong url, this page doesn't contain a video it contain a link to the actual video(http://www.bbcamerica.com/shows/the-hunt/full-episodes/season-1/episode-01-the-hardest-challenge), use the second link and you will be able to download the video, the video require TV provider auth, the only supported TV provider is DirecTV.
```
youtube-dl http://www.bbcamerica.com/shows/the-hunt/full-episodes/season-1/episode-01-the-hardest-challenge
[AMCNetworks] the-hardest-challenge: Downloading webpage
[AMCNetworks] the-hardest-challenge: Downloading JSON metadata
[AMCNetworks] 3Pjs118LbG1J: Downloading Provider Redirect Page
[AMCNetworks] 3Pjs118LbG1J: Downloading Provider Login Page
[AMCNetworks] 3Pjs118LbG1J: Logging in
[AMCNetworks] 3Pjs118LbG1J: Confirming Login
[AMCNetworks] 3Pjs118LbG1J: Retrieving Session
[AMCNetworks] 3Pjs118LbG1J: Retrieving Authorization Token
[AMCNetworks] 3Pjs118LbG1J: Retrieving Media Token
[AMCNetworks] 3Pjs118LbG1J: Downloading SMIL data
[AMCNetworks] 3Pjs118LbG1J: Downloading m3u8 information
WARNING: ar subtitles not available for 3Pjs118LbG1J
[info] Writing video subtitles to: The Hunt - Season 1 - The Hardest Challenge-3Pjs118LbG1J.en.vtt
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 254
[download] Destination: The Hunt - Season 1 - The Hardest Challenge-3Pjs118LbG1J.mp4
[download]   1.0% of ~803.05MiB at 351.82KiB/s ETA 47:31
```

**Comment by keybounce** [comment]
http://www.bbcamerica.com/shows/the-hunt/season-1/episode-01-the-hardest-challenge
http://www.bbcamerica.com/shows/the-hunt/full-episodes/season-1/episode-01-the-hardest-challenge

Ahh ... I had to look over those URL's a few times before I saw the difference.

Worse: They have Direct TV as a supported "cable" provider, but not Dish. Sigh.

So, I have to
1. Use a supported TV company
2. That company has to pay for full episode service
3. I have to have BBC America as part of my service.

I have BBCA. And I thought "fair use", including "time shifting" (the VCR case) would regard this as legal. But no, someone else has to pay them extra.

foo bar.

Thank you.
